pennylane/qchem/tapering.py

```diff
@@ -756,32 +761,38 @@ def taper_operation(
 
     # build generator for the operation either internally or using the provided op_gen
     op_gen = _build_generator(operation, wire_order, op_gen=op_gen)
-    # check compatibility between the generator and the symmeteries
-    if np.all(
-        [
-            [
-                qml.is_commuting(op1, op2)
-                for op1, op2 in itertools.product(generator.ops, op_gen.ops)
-            ]
-            for generator in generators
-        ]
-    ) and not np.all(np.isclose(op_gen.coeffs, np.zeros_like(op_gen.coeffs), rtol=1e-8)):
-        gen_tapered = qml.taper(op_gen, generators, paulixops, paulix_sector)
-    else:
-        gen_tapered = qml.Hamiltonian([], [])
-    gen_tapered = qml.simplify(gen_tapered)
+
+    # Performing commutation check for pauli sentences
+    def _is_commuting(ps1, ps2):
+        commutator = ps1.commutator(ps2)
+        commutator.simplify()
+        return commutator == PauliSentence({})
```

Comment:
Also need a TODO here to update this part when `qml.is_commuting` supports Pauli words and sentences.